### DISCUSSION TITLE
ERROR: "check-db" exited with 1.

### DISCUSSION THREAD

**Discussion by mariushosting** [umami-software/umami#1487] [General]
The latest umami update with the postgres latest database is not working same with mysql as database in some server not all.

yarn run v1.22.19
$ npm-run-all check-db update-tracker start-server
$ node scripts/check-db.js
✓ DATABASE_URL is defined.
✗ Unable to connect to the database.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "check-db" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
yarn run v1.22.19

**Comment by mikecao** [comment]
That step simply tries to connect to the database. Did you change your `DATABASE_URL` at all?

**Comment by mariushosting** [comment]
> That step simply tries to connect to the database. Did you change your `DATABASE_URL` at all?

Everything works perfectly. We only have a problem on some Synology Servers from the 20+ series 

Reference: https://mariushosting.com/how-to-install-umami-on-your-synology-nas/

The installation is smooth on all NAS devices, with the exception of 20+series on which it's still not working and can't connect to the db. I have received 50 reports in the last month from different users with 20+ series NAS. I don't know what is causing the issue, the only thing they have in common is the Intel Celeron J4125 CPU.

**Comment by tschannik** [comment]
@mariushosting i'm not familiar with your environment but i had the same issue today. For some reason my environment changes the `DATABASE_URL` from mysql://xxx to mysql2://xxx. 

Are you able to run commands on your system? Try printenv and verify the correct format of the `DATABASE_URL` env variable.

@mikecao is there an alternative solution to configure the connection? Like a way to not use a `DATABASE_URL` but configure the values seperately like `DATABASE_HOST,DATABASE_USER` etc?

**Comment by mikecao** [comment]
@tschannik Even if we split up the connection details, you'd still have to define environment variables. You'll have to figure out why your environment variables are not showing up.

@mariushosting Is it possible the 20+ has a firewall preventing external connections?

**Comment by tschannik** [comment]
@mikecao Well the problem is that in my environment i have no control over the `DATABASE_URL` env variable. It is automatically set by the system (even with the correct values) but it starts with mysql2:// instead of mysql://. So the script `check-db.js` verifies the existence of `DATABASE_URL` with is successful but the next step `prisma.$connect()` fails because of an unsupported url. 
Unfortunately i'm not able to rewrite the `DATABASE_URL` to sth else. To make it work i hat to create my own dockerfile with a customized entrypoint so i could "rename" the variable. 

My case is probably very specific. So i would agree that reading the values from different variables is not a "important" feature 👍

**Comment by mariushosting** [comment]
> 

The firewall settings is identical to other NAS. Only the 20+ models are getting trouble.

**Comment by mikecao** [comment]
@tschannik Can you try setting `DATABASE_TYPE`? That is another variable that we check.

**Comment by mikecao** [comment]
@mariushosting I only have a synology 19+ so I can't troubleshoot it

**Comment by wheeleran** [comment]
hey @mariushosting, i run umami without issue on a 1517+ using the postgres db. maybe try that?

here's my docker-compose:
`---
version: '3'
services:
  umami:
    container_name: umami
    image: ghcr.io/umami-software/umami:postgresql-latest
    ports:
      - "3000:3000"
    environment:
      DATABASE_URL: postgresql://umami:umami@db:5432/umami
      DATABASE_TYPE: postgresql
      HASH_SALT: hash-key-here
      PUID: puid-here
      PGID: admin-pgid-here
      TRACKER_SCRIPT_NAME: tracker-name-here
    depends_on:
      - db
    restart: always
  db:
    container_name: umami-db
    image: postgres:12-alpine
    environment:
      POSTGRES_DB: umami
      POSTGRES_USER: umami
      POSTGRES_PASSWORD: umami
    volumes:
      - /volume1/docker/umami/sql/schema.postgresql.sql:/docker-entrypoint-initdb.d/schema.postgresql.sql:ro
      - /volume1/docker/umami/umami-db-data:/var/lib/postgresql/data
    restart: always
volumes:
  umami-db-data: 
`

big fan of your work!

**Comment by mariushosting** [comment]
> hey @mariushosting, i run umami without issue on a 1517+ using the postgres db. maybe try that?
> 
> here's my docker-compose: `--- version: '3' services: umami: container_name: umami image: ghcr.io/umami-software/umami:postgresql-latest ports: - "3000:3000" environment: DATABASE_URL: postgresql://umami:umami@db:5432/umami DATABASE_TYPE: postgresql HASH_SALT: hash-key-here PUID: puid-here PGID: admin-pgid-here TRACKER_SCRIPT_NAME: tracker-name-here depends_on: - db restart: always db: container_name: umami-db image: postgres:12-alpine environment: POSTGRES_DB: umami POSTGRES_USER: umami POSTGRES_PASSWORD: umami volumes: - /volume1/docker/umami/sql/schema.postgresql.sql:/docker-entrypoint-initdb.d/schema.postgresql.sql:ro - /volume1/docker/umami/umami-db-data:/var/lib/postgresql/data restart: always volumes: umami-db-data: `
> 
> big fan of your work!

Hello wheeleran, as already write at the beginning of this post the problem is only with NAS series model x20 not x17.

**Comment by stianlagstad** [comment]
In case this is the issue for anyone else reading this thread: I discovered that if my `DATABASE_URL` has specific characters, such as the `#` character in a password, then `yarn check-db` fails:
```bash
yarn run v1.22.19
$ node scripts/check-db.js
✓ DATABASE_URL is defined.
✗ Unable to connect to the database.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Reply by MattFaz** [reply]
Thank you @stianlagstad - my issues was a special character `#` in the password.

**Comment by Jimmy-Effe** [comment]
`✓ DATABASE_URL is defined.
✓ Database connection successful.
✓ Database version check successful.
Error: P3009

migrate found failed migrations in the target database, new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
The `02_report_schema_session_data` migration started at 2023-08-25 14:45:56.100 UTC failed


✗ Command failed: prisma migrate deploy
Error: P3009

migrate found failed migrations in the target database, new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
The `02_report_schema_session_data` migration started at 2023-08-25 14:45:56.100 UTC failed



ERROR: "check-db" exited with 1.`
Maybe this issue is not to fix in [2683ff2]

**Reply by Jimmy-Effe** [reply]
i can connect database, but i don't know why error by check-db

**Comment by LTK-Leo** [comment]
I had the same issue and mine was due to SSL/TLS. I had to set the MySQL server's parameter "require_secure_transport" OFF.

**Comment by teledyn** [comment]
I do have require_secure_transport set to OFF, with GRANT ALL permissions for the user@localhost, and the same user and pw works with mysql -u -p; using urls with mysql://umami:<PW>@localhost:3306/umami ('umami' is also the db name)

pw contains no special characters, and I've tried both mysql and mysql2
```
$ node scripts/check-db.js
✓ DATABASE_URL is defined.
✗ Unable to connect to the database.
```
I'm using the current git code with Debian 12 on a rpi-4B+ with mysql  15.1 (10.11.4-MariaDB)

There is just one clue in the log:
```
2024-03-11T17:19:15.501240-04:00 myhost mariadbd[6836]: 2024-03-11 17:19:15 3360073 [Warning] Aborted connection 3360073 to db: 'unconnected' user: 'unauthenticated' host: '127.0.0.1' (This connection closed normally without authentication)
```
but again, using mysql from the command line with that username and pw logs in correctly, however if I specify -P 3306, _then_ I get
```
2024-03-11T17:19:15.501240-04:00 nextcloudpi mariadbd[6836]: 2024-03-11 17:19:15 3360073 [Warning] Aborted connection 3360073 to db: 'unconnected' user: 'unauthenticated' host: '127.0.0.1' (This connection closed normally without authentication)
```
and here's the answer!
```
CREATE USER 'umami'@'127.0.0.1' IDENTIFIED BY 'PW';
GRANT ALL ON umami.* TO umami@'127.0.0.1';
FLUSH PRIVILEGES;
```
and now it works:
```
$ node scripts/check-db.js
✓ DATABASE_URL is defined.
✓ Database connection successful.
✓ Database version check successful.
```

**Comment by ENTPRESTIGIOUS** [comment]
I used the `docker-compose.yml` and I faced the same issue. I fixed it by mounting the database into a native local directory.

**Comment by OskarPetersilie** [comment]
In my case docker container wasnt able to call internet.

I used "iptables:false" in /etc/docker/deamon.json whcih is BAD practice. rather use  https://github.com/chaifeng/ufw-docker.

Worked after that fix was made

```bash
docker compose logs umami --tail 10
umami-1  | ✓ DATABASE_URL is defined.
umami-1  | ✓ Database connection successful.
umami-1  | ✓ Database version check successful.
umami-1  | Error: request to https://binaries.prisma.sh/all_commits/34b5a692b7bd79939a9a2c3ef97d816e749cda2f/linux-musl-openssl-3.0.x/schema-engine.gz.sha256 failed, reason:
umami-1  | ✗ Command failed: prisma migrate deploy
umami-1  | Error: request to https://binaries.prisma.sh/all_commits/34b5a692b7bd79939a9a2c3ef97d816e749cda2f/linux-musl-openssl-3.0.x/schema-engine.gz.sha256 failed, reason:
umami-1  |
umami-1  |  ELIFECYCLE  Command failed with exit code 1.
umami-1  | ERROR: "check-db" exited with 1.
umami-1  |  ELIFECYCLE  Command failed with exit code 1.
```
